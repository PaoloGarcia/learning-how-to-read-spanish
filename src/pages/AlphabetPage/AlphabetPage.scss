.AlphabetPage {
    width: 100%;

    .ant-tabs-tabpane {
        display: flex;
        flex-wrap: wrap;
        justify-content: center;
        overflow-y: scroll;
    }
}
