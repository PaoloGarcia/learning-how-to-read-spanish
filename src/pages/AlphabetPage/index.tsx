import { Tabs } from "antd";
import Layout from "../../components/Layout";
import Card from "../../components/Card";
import { alphabet, consonants, vowels } from "../../constants";
import "./AlphabetPage.scss";

function AlphabetPage(): JSX.Element {
    const renderCards = (LettersArr: string[]): JSX.Element[] => {
        return LettersArr.map(
            (letter: string) => <Card key={letter} text={letter} />
        );
    };

    return (
        <Layout backButton title="Abecedario">
            <div className="AlphabetPage">
                <Tabs defaultActiveKey="1" centered size="small">
                    <Tabs.TabPane tab="Todas" key="1">
                        {renderCards(alphabet)}
                    </Tabs.TabPane>
                    <Tabs.TabPane tab="Vocales" key="2">
                        {renderCards(vowels)}
                    </Tabs.TabPane>
                    <Tabs.TabPane tab="Consonantes" key="3">
                        {renderCards(consonants)}
                    </Tabs.TabPane>
                </Tabs>
            </div>
        </Layout>
    );
}

export default AlphabetPage;
