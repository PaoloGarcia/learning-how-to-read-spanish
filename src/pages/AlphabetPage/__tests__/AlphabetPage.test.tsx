import React from "react"
import { render } from "@testing-library/react"
import AlphabetPage from ".."

describe("Snapshot Testing", () => {
    test("rendering component with no args", () => {
        const { asFragment } = render(<AlphabetPage />)
        const firstRender: DocumentFragment = asFragment()
        expect(firstRender).toMatchSnapshot()
    })
})