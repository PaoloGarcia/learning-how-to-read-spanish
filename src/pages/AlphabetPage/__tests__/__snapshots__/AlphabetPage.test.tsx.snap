// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing rendering component with no args 1`] = `
<DocumentFragment>
  <div
    class="Layout"
  >
    <nav
      class="back-btn"
    >
      <button
        class="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
        type="button"
      >
        <span
          aria-label="arrow-left"
          class="anticon anticon-arrow-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="arrow-left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
      </button>
    </nav>
    <header>
      <h1>
        Abecedario
      </h1>
    </header>
    <main>
      <div
        class="AlphabetPage"
      >
        <div
          class="ant-tabs ant-tabs-top ant-tabs-small ant-tabs-centered"
        >
          <div
            class="ant-tabs-nav"
            role="tablist"
          >
            <div
              class="ant-tabs-nav-wrap"
            >
              <div
                class="ant-tabs-nav-list"
                style="transform: translate(0px, 0px);"
              >
                <div
                  class="ant-tabs-tab ant-tabs-tab-active"
                >
                  <div
                    aria-controls="rc-tabs-test-panel-1"
                    aria-selected="true"
                    class="ant-tabs-tab-btn"
                    id="rc-tabs-test-tab-1"
                    role="tab"
                    tabindex="0"
                  >
                    Todas
                  </div>
                </div>
                <div
                  class="ant-tabs-tab"
                >
                  <div
                    aria-controls="rc-tabs-test-panel-2"
                    aria-selected="false"
                    class="ant-tabs-tab-btn"
                    id="rc-tabs-test-tab-2"
                    role="tab"
                    tabindex="0"
                  >
                    Vocales
                  </div>
                </div>
                <div
                  class="ant-tabs-tab"
                >
                  <div
                    aria-controls="rc-tabs-test-panel-3"
                    aria-selected="false"
                    class="ant-tabs-tab-btn"
                    id="rc-tabs-test-tab-3"
                    role="tab"
                    tabindex="0"
                  >
                    Consonantes
                  </div>
                </div>
                <div
                  class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                />
              </div>
            </div>
            <div
              class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
            >
              <button
                aria-controls="rc-tabs-test-more-popup"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-hidden="true"
                class="ant-tabs-nav-more"
                id="rc-tabs-test-more"
                style="visibility: hidden; order: 1;"
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="ellipsis"
                  class="anticon anticon-ellipsis"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="ellipsis"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
          <div
            class="ant-tabs-content-holder"
          >
            <div
              class="ant-tabs-content ant-tabs-content-top"
            >
              <div
                aria-hidden="false"
                aria-labelledby="rc-tabs-test-tab-1"
                class="ant-tabs-tabpane ant-tabs-tabpane-active"
                id="rc-tabs-test-panel-1"
                role="tabpanel"
                tabindex="0"
              >
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Aa
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Bb
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Cc
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Dd
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Ee
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Ff
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Gg
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Hh
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Ii
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Jj
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Kk
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Ll
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Mm
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Nn
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Ññ
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Oo
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Pp
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Qq
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Rr
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Ss
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Tt
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Uu
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Vv
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Ww
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Xx
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Yy
                    </span>
                  </button>
                </div>
                <div
                  class="Card"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round ant-btn-sm ant-btn-background-ghost"
                    type="button"
                  >
                    <span>
                      Zz
                    </span>
                  </button>
                </div>
              </div>
              <div
                aria-hidden="true"
                aria-labelledby="rc-tabs-test-tab-2"
                class="ant-tabs-tabpane"
                id="rc-tabs-test-panel-2"
                role="tabpanel"
                style="display: none;"
                tabindex="-1"
              />
              <div
                aria-hidden="true"
                aria-labelledby="rc-tabs-test-tab-3"
                class="ant-tabs-tabpane"
                id="rc-tabs-test-panel-3"
                role="tabpanel"
                style="display: none;"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
